src/emu.c

```diff
@@ -255,12 +269,13 @@ static int emu_client_event_cb_emp (EmuClient *client, const char *eventType, co
     sentProgress
   );
 
-  // TODO: Check better remaining value.
-  if (
-    iterationValue > 0 &&
-    (remainingValue <= 50 || iterationValue >= 4) &&
-    client->emu->state != EMU_STATE_LIVE_STAGE_DONE
-  ) {
+  if (iterationValue <= 0 || client->emu->state == EMU_STATE_LIVE_STAGE_DONE)
+    return 0;
+
+  if ((iterationValue >= MIGRATION_IT_GRACE_COUNT && (
+    progress->iterationWithoutProgress >= MIGRATION_IT_MAX_WITHOUT_PROGRESS ||
+    iterationValue >= MIGRATION_IT_MAX
+  )) || (remainingValue >= 0 && remainingValue <= 16384)) {
```

Comment:
Why 16384?

Reply:
Not totally arbitrary, we determined this value with Damien by looking at the average time taken to migrate the remaining RAM pages on various pools. We estimate that 16384 is a low enough value to be migrated in less than a second. Also it is difficult to use a smaller value if there is a lot of activity on the machine.
On the other hand it should be in a #define.

Also we must see this value as a limit since we will take into account the transfer rate in a second commit. It should be more precise.

Reply:
> On the other hand it should be in a #define.

It was my point a define with a clear name :)